database/buildSchema/43_views_functions_triggers.sql

```diff
@@ -308,11 +281,30 @@ ALTER TABLE public.template_element
 
 ALTER TABLE public.template_element
     DROP COLUMN IF EXISTS template_version,
-    ADD COLUMN IF NOT EXISTS template_version integer GENERATED ALWAYS AS (public.get_template_version (section_id)) STORED;
+    ADD COLUMN IF NOT EXISTS template_version varchar GENERATED ALWAYS AS (public.get_template_version (section_id)) STORED;
 
 ALTER TABLE public.template_element
     ADD UNIQUE (template_code, code, template_version);
 
+-- FUNCTION/TRIGGER to re-compute the above generated values whenever the versionId or code changes
+CREATE OR REPLACE FUNCTION public.recompute_template_element_code_version ()
+    RETURNS TRIGGER
+    AS $template_element_event$
+BEGIN
+    UPDATE public.template_element
+        SET code=code
+        WHERE template_code = OLD.code;
+        RETURN NULL;
+END;
+$template_element_event$
+LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS template_version_code_update ON public.template;
+CREATE TRIGGER template_version_code_update
+    AFTER UPDATE OF version_id, code ON public.template
+    FOR EACH ROW
+    EXECUTE FUNCTION public.recompute_template_element_code_version ();
```

Comment:
This was causing a problem -- when I updated the "versionId" when "committing" a version, the generated columns on `template_element` weren't getting updated, which resulted in database errors when a new version used the "*" placeholder versionId. So this just ensures that those values get recomputed whenever the versionId or code are updated in the template table.